runtime/doc/msgpack_rpc.txt

```diff
@@ -242,4 +242,170 @@ Even for statically compiled clients it is good practice to avoid hardcoding
+
+	{buf} is an API handle for the buffer.
+
+  {changedtick} is the value of |b:changedtick| for the buffer. If you send an
+  API command back to neovim you can check the value of |b:changedtick| as
+  part of your request to ensure that no other changes have been made.
+
+  {firstline} is the integer line number of the first line that was replaced.
+  Note that {firstline} is zero-indexed, so if line `1` was replaced then
+  {firstline} will be `0` instead of `1`. {firstline} is guaranteed to always
+  be less than or equal to the number of lines that were in the buffer before
+  the lines were replaced.
+
+  {lastline} is the integer line number of the first line that was not replaced
+  (i.e. the range {firstline}, {lastline} is end-exclusive). Note that
+  {lastline} is zero-indexed, so if line numbers 2 to 5 were replaced, this
+  will be `5` instead of `6`. {lastline} is guaranteed to always be less than
+  or equal to the number of lines that were in the buffer before the lines were
+  replaced.
+
+  {linedata} is a list of strings containing the contents of the new buffer
+  lines. Newline characters are not included in the strings, so empty lines
+  will be given as empty strings. 
+
+  Note: sometimes {changedtick} will be |v:null|, which means that the buffer
+  text *looks* like it has changed, but actually hasn't. In this case the lines
+  in {linedata} contain the modified text that is shown to the user, but
+  doesn't reflect the actual buffer contents. Currently this behaviour is
+  only used for the 'inccommand' option.
+
+nvim_buf_update_tick[{buf}, {changedtick}]		                   *nvim_buf_update_tick*
+
+  Indicates that |b:changedtick| was incremented for the buffer {buf}, but no
+  text was changed. This is currently only used by undo/redo.
+
+  {buf} is an API handle for the buffer.
+
+  {changedtick} is the new value of |b:changedtick| for that buffer.
+
+nvim_buf_updates_end[{buf}]                                    		*nvim_buf_updates_end*
+
+  {buf} is an API handle for the buffer.
+
+  Indicates that buffer updates for the nominated buffer have been disabled,
+  either by calling |nvim_buf_updates_stop| or because the buffer was unloaded
+  (see |buffer-updates-limitations| for more information).
+
+                                                      *buffer-updates-limitations*
+Limitations~
+
+Note that any of the following actions will also turn off buffer updates because
+the buffer contents are unloaded from memory:
+
+  - Closing all a buffer's windows (unless 'hidden' is enabled).
+  - Using |:edit| to reload the buffer
+  - reloading the buffer after it is changed from outside neovim.
+
+                                                       *buffer-updates-examples*
+Example Events~
+
+If buffer updates are activated a new empty buffer, the following
+|nvim_buf_updates_start| event will be sent: >
+
+  rpcnotify({channel}, "nvim_buf_updates_start", [{buf}, [""], v:false])
```

Comment:
also don't use `rpcnotify` here.

Reply:
Right, I removed that and just gave the events as examples.